sdks/python/apache_beam/options/pipeline_options_validator_test.py

```diff
@@ -95,258 +96,152 @@ def test_missing_required_options(self):
           '--staging_location=gs://foo/bar',
           '--temp_location=gs://foo/bar'
       ]
 
-      if num_workers is not None:
-        options.append('--num_workers=' + num_workers)
+      if _num_workers is not None:
+        options.append('--num_workers=' + _num_workers)
 
       pipeline_options = PipelineOptions(options)
       runner = MockRunners.DataflowRunner()
       validator = PipelineOptionsValidator(pipeline_options, runner)
       return validator
 
-    test_cases = [
-        {
-            'num_workers': None, 'errors': []
-        },
-        {
-            'num_workers': '1', 'errors': []
-        },
-        {
-            'num_workers': '0', 'errors': ['num_workers']
-        },
-        {
-            'num_workers': '-1', 'errors': ['num_workers']
-        },
-    ]
-
-    for case in test_cases:
-      errors = get_validator(case['num_workers']).validate()
-      self.assertEqual(
-          self.check_errors_for_arguments(errors, case['errors']), [])
-
-  def test_is_service_runner(self):
-    test_cases = [
-        {
-            'runner': MockRunners.OtherRunner(),
-            'options': [],
-            'expected': False,
-        },
-        {
-            'runner': MockRunners.OtherRunner(),
-            'options': ['--dataflow_endpoint=https://dataflow.googleapis.com'],
-            'expected': False,
-        },
-        {
-            'runner': MockRunners.OtherRunner(),
-            'options': ['--dataflow_endpoint=https://dataflow.googleapis.com/'],
-            'expected': False,
-        },
-        {
-            'runner': MockRunners.DataflowRunner(),
-            'options': [],
-            'expected': True,
-        },
-        {
-            'runner': MockRunners.DataflowRunner(),
-            'options': ['--dataflow_endpoint=https://another.service.com'],
-            'expected': True,
-        },
-        {
-            'runner': MockRunners.DataflowRunner(),
-            'options': ['--dataflow_endpoint=https://dataflow.googleapis.com'],
-            'expected': True,
-        },
-        {
-            'runner': MockRunners.DataflowRunner(),
-            'options': ['--dataflow_endpoint=http://localhost:1000'],
-            'expected': False,
-        },
-        {
-            'runner': MockRunners.DataflowRunner(),
-            'options': ['--dataflow_endpoint=foo: //dataflow. googleapis. com'],
-            'expected': True,
-        },
-        {
-            'runner': MockRunners.DataflowRunner(),
-            'options': [],
-            'expected': True,
-        },
-    ]
-
-    for case in test_cases:
-      validator = PipelineOptionsValidator(
-          PipelineOptions(case['options']), case['runner'])
-      self.assertEqual(validator.is_service_runner(), case['expected'], case)
+    errors = get_validator(num_workers).validate()
+    self.assertEqual(
+        self.check_errors_for_arguments(errors, expected_error_args), [])
+
+  @parameterized.expand([
```

Comment:
Nothing blocking but I like this way of writing test cases (each test case and its param on different line).. could we do this for other test cases as well
```
[
  (
      [],
      {},
   ),(
      [],
      {},
   ),
]
```

Reply:
I've tried to have it on different lines, but the autoformatting reverted the changes, maybe worth adjusting the formatter?

Reply:
ohh I see, thanks! looks good to merge.